mmv1/products/dialogflowcx/SecuritySettings.yaml

```diff
@@ -0,0 +1,161 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'SecuritySettings'
+base_url: 'projects/{{project}}/locations/{{location}}/securitySettings'
+update_verb: :PATCH
+update_mask: true
+description: |
+  Represents the settings related to security issues, such as data redaction and data retention. It may take hours for updates on the settings to propagate to all the related components and take effect.
+  Multiple security settings can be configured in each location. Each agent can specify the security settings to apply, and each setting can be applied to multiple agents in the same project and location.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/dialogflow/cx/docs'
+  api: 'https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.securitySettings'
+timeouts: !ruby/object:Api::Timeouts
+  insert_minutes: 40
+  update_minutes: 40
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dialogflowcx_security_settings_basic'
+    primary_resource_id: 'basic_security_settings'
+    vars:
+      settings_name: 'dialogflowcx-security-settings'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dialogflowcx_security_settings_full'
+    primary_resource_id: 'basic_security_settings'
+    vars:
+      inspect_name: 'dialogflowcx-inspect-template'
+      deidentify_name: 'dialogflowcx-deidentify-template'
+      settings_name: 'dialogflowcx-security-settings'
+      bucket_name: 'dialogflowcx-bucket'
+    test_env_vars:
+      project: :PROJECT_NAME
+id_format: 'projects/{{project}}/locations/{{location}}/securitySettings/{{name}}'
+import_format: ['projects/{{project}}/locations/{{location}}/securitySettings/{{name}}']
+skip_sweeper: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The unique identifier of the settings.
+      Format: projects/<Project ID>/locations/<Location ID>/securitySettings/<Security Settings ID>.
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      The location these settings are located in.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      The human-readable name of the security settings, unique within the location.
+    required: true
+  - !ruby/object:Api::Type::Enum
+    name: 'redactionStrategy'
+    description: |
+      Defines how we redact data.
```

Comment:
Consider noting the behaviour when the field is omitted i.e. "If omitted or set to "", do not redact data."

Reply:
(same with other single-value enums)

Reply:
Good call, I re-added the enum 0-values and updated the documentation to specify the default behavior for `redactionStrategy` and `redactionScope`. I found some places that had `default_value: THE_ENUM_0_VALUE` but it's not consistent and I'm nervous about possible breaking changes in the future -> I didn't do it here.

I didn't touch `purgeDataTypes` because it's an array of enums and it's obviously nonsense to send `["PURGE_DATA_TYPE_UNSPECIFIED"]` or add `AUDIO_FORMAT_UNSPECIFIED` for the same reason. And I didn't touch `retentionStrategy` because it conflicts with `retentionWindowDays` -> there's no meaningful 0-value.

WDYT?

Reply:
Fair enough; I've seen it both ways and don't personally have a preference but this might just be because I'm used to it by now 😅.